pkg/controllers/edgeconnect/phase_test.go

```diff
@@ -0,0 +1,74 @@
+package edgeconnect
+
+import (
+	"testing"
+
+	"github.com/Dynatrace/dynatrace-operator/pkg/api/scheme/fake"
+	"github.com/Dynatrace/dynatrace-operator/pkg/api/status"
+	"github.com/Dynatrace/dynatrace-operator/pkg/api/v1alpha2/edgeconnect"
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func TestEdgeConnectPhaseChanges(t *testing.T) {
+	ec := &edgeconnect.EdgeConnect{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      testName,
+			Namespace: testNamespace,
+		},
+		Spec: edgeconnect.EdgeConnectSpec{},
+	}
+
+	t.Run("no edgeConnect deployment in cluster -> deploying", func(t *testing.T) {
+		fakeClient := fake.NewClient()
+		controller := &Controller{
+			client:    fakeClient,
+			apiReader: fakeClient,
+		}
+		phase := controller.determineEdgeConnectPhase(ec)
+		assert.Equal(t, status.Deploying, phase)
+	})
+
+	t.Run("error accessing k8s api -> error", func(t *testing.T) {
+		fakeClient := errorClient{}
+		controller := &Controller{
+			client:    fakeClient,
+			apiReader: fakeClient,
+		}
+		phase := controller.determineEdgeConnectPhase(ec)
+		assert.Equal(t, status.Error, phase)
+	})
+
+	t.Run("edgeConnect pods not ready -> deploying", func(t *testing.T) {
+		replicas, readyReplicas := int32(1), int32(0)
+		objects := []client.Object{
+			createDeployment(testNamespace, "test-name", replicas, readyReplicas),
+		}
+
+		fakeClient := fake.NewClient(objects...)
+
+		controller := &Controller{
+			client:    fakeClient,
+			apiReader: fakeClient,
+		}
+		phase := controller.determineEdgeConnectPhase(ec)
+		assert.Equal(t, status.Deploying, phase)
+	})
+
+	t.Run("edgeConnect deployed -> running", func(t *testing.T) {
+		replicas, readyReplicas := int32(1), int32(1)
+		objects := []client.Object{
+			createDeployment(testNamespace, "test-name", replicas, readyReplicas),
+		}
+
+		fakeClient := fake.NewClient(objects...)
+
+		controller := &Controller{
+			client:    fakeClient,
+			apiReader: fakeClient,
+		}
+		phase := controller.determineEdgeConnectPhase(ec)
```

Comment:
ec.Spec.Replicas=0 so generally speaking  the test is broken

Reply:
I don't get it, check line number 60-61. replicas == readyReplicas == 1

Reply:
ec.Spec.Replicas is NIL so according to the `determineEdgeConnectPhase()`
```
scheduledReplicas := int32(0)
	if ec.Spec.Replicas != nil {
		scheduledReplicas = *deployment.Spec.Replicas
	}

	if scheduledReplicas != deployment.Status.ReadyReplicas {
```	
the fact that `deployment.replicas` = `deployment.readyReplicas` = 1 has nothing to do with the correct logic and final result

Reply:
fixed in 93c4f1d0